core/src/main/scala_2.13-/shapeless/LazyMacros.scala

```diff
@@ -165,14 +165,7 @@ class LazyMacros(val c: whitebox.Context) extends CaseClassMacros with OpenImpli
           case Right((state, inst)) =>
             val (tree, actualType) = if (root) mkInstances(state)(instTpe0) else (inst.ident, inst.actualTpe)
             current = if (root) None else Some(state)
-            if (root) {
-              val valNme = TermName(c.freshName("inst"))
-              q"""
-              val $valNme: $actualType = $tree
```

Comment:
@milessabin why was this an eager val? It caused execution order tests to fail:
https://travis-ci.org/github/milessabin/shapeless/jobs/695762315